File: Dev/C++/Mascaret/src/VEHA/Behavior/BasicActions/CallOperationBehaviorExecution.cpp
#include "VEHA/Behavior/BasicActions/CallOperationBehaviorExecution.h"
#include "VEHA/Kernel/InstanceSpecification.h"
#include "VEHA/Kernel/Operation.h"
#include <boost/bind.hpp>


namespace VEHA
{

CallOperationBehaviorExecution::CallOperationBehaviorExecution(shared_ptr<CallOperationAction> action, shared_ptr<InstanceSpecification> host, const Parameters& p):BehaviorExecution(action,host,p), _action(action)
{
	_behaviorExecution=_action->getOperation()->getMethod()->createBehaviorExecution(getHost(),parameters);
}

CallOperationBehaviorExecution::~CallOperationBehaviorExecution()
{
}

void CallOperationBehaviorExecution::stop()
{
	BehaviorExecution::stop();
	_behaviorExecution->stop();
}
void CallOperationBehaviorExecution::restart()
{
	BehaviorExecution::restart();
	_behaviorExecution->restart();
}
void CallOperationBehaviorExecution::pause()
{
	BehaviorExecution::pause();
	_behaviorExecution->pause();
}

double CallOperationBehaviorExecution::execute(double dt)
{
	return _behaviorExecution->execute(dt);
}

} // Namespace

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/BasicActions/CallOperationBehaviorExecution.cs
using System.Collections;
using System.Collections.Generic;

namespace Mascaret
{
    public class CallOperationBehaviorExecution : BehaviorExecution
    {
        public CallOperationAction action;
        public BehaviorExecution behaviorExecution;


        public CallOperationBehaviorExecution(CallOperationAction paction, InstanceSpecification host, Dictionary<string, ValueSpecification> p)
            : base(paction, host, p, false)
        {
            this.action = paction;
            MascaretApplication.Instance.VRComponentFactory.Log("CallOperationAction : " + action.Operation.Method);

            foreach (ValuePin pin in action.ValuePins)
            {
                p.Add(pin.name, pin.ValueSpec);
            }
            MascaretApplication.Instance.VRComponentFactory.Log("READY TO Start");

            behaviorExecution = action.Operation.Method.createBehaviorExecution(this.Host, p, false);
            if (behaviorExecution == null) MascaretApplication.Instance.VRComponentFactory.Log("Chérie ca va trancher");
  
        }

        public override void stop()
        {
            base.stop();
            behaviorExecution.stop();
        }

        public override void restart()
        {
            base.restart();
            behaviorExecution.restart();
        }

        public override void pause()
        {
            base.pause();
            behaviorExecution.pause();
        }

        public override double execute(double dt)
        {
            return behaviorExecution.execute(dt);
        }
    }
}
